src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoroelasticKernel.hpp

```diff
@@ -0,0 +1,447 @@
+    real64 const volumetricStrainOld = volumetricStrainNew - FE_TYPE::symmetricGradientTrace( dNdX, stack.uhat_local );//
+    real64 const porosityOld = m_poroRef( k ) + m_biotCoefficient * volumetricStrainOld;// +  DeltaPoro                //
+    real64 const dPorosity_dPressure = biotSkeletonModulusInverse;                                                     //
+    real64 const dPorosity_dVolStrainIncrement =  m_biotCoefficient;                                                   //
+                                                                                                                       //
+    GEOSX_ERROR_IF_GT_MSG( fabs( m_biotCoefficient - 1.0 ),                                                            //
+                           1e-10,                                                                                      //
+                           "Correct only for Biot's coefficient equal to 1" );                                         //
+    // --------------------------------------------------------------------------------------------------------------- //
+    real64 const porosityNew = porosityOld
+                               + m_biotCoefficient * (strainIncrement[0] + strainIncrement[1] + strainIncrement[2] )
+                               + biotSkeletonModulusInverse * m_deltaFluidPressure[k];
+
+
+    // Evaluate body force vector
+    real64 bodyForce[3] = { m_gravityVector[0],
+                            m_gravityVector[1],
+                            m_gravityVector[2]};
+    if( m_gravityAcceleration > 0.0 )
+    {
+      real64 mixtureDensity = ( 1.0 - porosityNew ) * m_solidDensity( k, q ) + porosityNew * m_fluidDensity( k, q );
+      mixtureDensity *= detJxW;
+      bodyForce[0] *= mixtureDensity;
+      bodyForce[1] *= mixtureDensity;
+      bodyForce[2] *= mixtureDensity;
+    }
+
+    // Assemble local jacobian and residual
+
+    // --- Momentum balance
+    for( localIndex i=0; i<6; ++i )
+    {
+      totalStress[i] *= -detJxW;
+    }
+
+    FE_TYPE::plusGradNajAijPlusNaFi( dNdX,
+                                     totalStress,
+                                     N,
+                                     bodyForce,
+                                     reinterpret_cast< real64 (&)[numNodesPerElem][3] >(stack.localResidual) );
+
+    stiffness.template upperBTDB< numNodesPerElem >( dNdX, -detJxW, stack.localJacobian );
+
+    for( integer a = 0; a < numNodesPerElem; ++a )
+    {
+      stack.localDispFlowJacobian[ a * 3 + 0][0] += dNdX[a][0] * m_biotCoefficient * detJxW;
+      stack.localDispFlowJacobian[ a * 3 + 1][0] += dNdX[a][1] * m_biotCoefficient * detJxW;
+      stack.localDispFlowJacobian[ a * 3 + 2][0] += dNdX[a][2] * m_biotCoefficient * detJxW;
+    }
+
+    if( m_gravityAcceleration > 0.0 )
+    {
+      // Assumptions: ( i) dMixtureDens_dVolStrain contribution is neglected
+      //              (ii) grains are assumed incompressible
+
```

Comment:
These assumptions are just temporal?

Reply:
Yes, these are temporary assumptions. However, we should preserve the flexibility of lagging in time some derivatives in the linearization. Note that as soon as `dMixtureDens_dVolStrain` effects are accounted for the "displacement-displacement" block in the Jacobian matrix will result in a nonsymmetric block. As to grain compressibility effects, we should add a bulk modulus member -- or better a bulk compressibility for easily accounting for the incompressible case -- in the base class for all solid constitutive models.